src/rust/src/encoder/g608.rs

```diff
@@ -0,0 +1,308 @@
+    let mut m1: u32 = 0;
+    let mut s1: u32 = 0;
+    let mut ms1: u32 = 0;
+
+    let mut h2: u32 = 0;
+    let mut m2: u32 = 0;
+    let mut s2: u32 = 0;
+    let mut ms2: u32 = 0;
+
+    unsafe {
+        ccxr_millis_to_time(data.start_time, &mut h1, &mut m1, &mut s1, &mut ms1);
+        ccxr_millis_to_time(data.end_time - 1, &mut h2, &mut m2, &mut s2, &mut ms2);
+    }
+
+    // Increment counter
+    context.srt_counter += 1;
+
+    // Create timeline string for counter
+    let counter_line = format!("{}{}", context.srt_counter, unsafe {
+        std::str::from_utf8_unchecked(std::slice::from_raw_parts(
+            context.encoded_crlf.as_ptr(),
+            context.encoded_crlf_length as usize,
+        ))
+    });
+
+    // Encode and write counter line
+    let buffer_slice =
+        unsafe { std::slice::from_raw_parts_mut(context.buffer, context.capacity as usize) };
+    let used = encode_line(context, buffer_slice, counter_line.as_bytes());
+
+    if write_wrapped(unsafe { (*context.out).fh }, &buffer_slice[..used as usize]).is_err() {
+        return 0;
+    }
+
+    // Create timeline string for timestamps
+    let timestamp_line = format!(
+        "{:02}:{:02}:{:02},{:03} --> {:02}:{:02}:{:02},{:03}{}",
+        h1,
+        m1,
+        s1,
+        ms1,
+        h2,
+        m2,
+        s2,
+        ms2,
+        unsafe {
+            std::str::from_utf8_unchecked(std::slice::from_raw_parts(
+                context.encoded_crlf.as_ptr(),
+                context.encoded_crlf_length as usize,
+            ))
+        }
+    );
+
+    // Encode and write timestamp line
+    let used = encode_line(context, buffer_slice, timestamp_line.as_bytes());
+
+    if write_wrapped(unsafe { (*context.out).fh }, &buffer_slice[..used as usize]).is_err() {
+        return 0;
+    }
+
+    // Write all 15 lines with their encoding information
+    for i in 0..15 {
+        let subline_slice = unsafe {
+            std::slice::from_raw_parts_mut(context.subline, 1024) // temporary, should be inputted after encoder_ctx => EncoderCtx
+        };
+
+        // Get line encoded
+        let length = get_line_encoded(context, subline_slice, i, data);
+        if write_wrapped(
+            unsafe { (*context.out).fh },
+            &subline_slice[..length as usize],
+        )
+        .is_err()
+        {
+            return 0;
+        }
+
+        // Get color encoded
+        let length = get_color_encoded(context, subline_slice, i, data);
+        if write_wrapped(
+            unsafe { (*context.out).fh },
+            &subline_slice[..length as usize],
+        )
+        .is_err()
+        {
+            return 0;
+        }
+
+        // Get font encoded
+        let length = get_font_encoded(context, subline_slice, i, data);
+        if write_wrapped(
+            unsafe { (*context.out).fh },
+            &subline_slice[..length as usize],
+        )
+        .is_err()
+        {
+            return 0;
+        }
```

Comment:
We can simplify this one as well:
```suggestion
	let line_length = get_line_encoded(context, subline_slice, i, data);
	let color_length = get_color_encoded(context, subline_slice, i, data);
	let font_length = get_font_encoded(context, subline_slice, i, data);
		
	let lengths = [line_length, color_length, font_length];
	for j in 0..3 {
	    if write_wrapped(
	        unsafe { (*context.out).fh },
	        &subline_slice[..lengths[j] as usize],
	    ).is_err() { return 0; }
	}
```